core/src/mender-client.c

```diff
@@ -57,6 +57,11 @@
 #define CONFIG_MENDER_CLIENT_UPDATE_POLL_INTERVAL (1800)
 #endif /* CONFIG_MENDER_CLIENT_UPDATE_POLL_INTERVAL */
 
+/**
+ * @brief Deployment data version number (allows us to add/remove/modify fields in the future)
+ */
+#define MENDER_CLIENT_DEPLOYMENT_DATA_VERSION 1
```

Comment:
I don't think this belong here. I really should be part of `mender-storage.h` and do the sanity check encapsulated in a function there. But then, we are already exposing all the fields of the state data in the `mender-client` code, so maybe we can just let it be :thinking: